libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -457,67 +222,46 @@ - (NSSet*)allExistingAccountNames {
 
 /** Returns a unique user account identifier
  */
-- (NSString*)uniqueUserAccountIdentifier {
+- (NSString*)uniqueUserAccountIdentifier:(NSString *)clientId {
     NSSet *existingAccountNames = [self allExistingAccountNames];
-   
+
     // Make sure to build a unique identifier
     NSString *identifier = nil;
     while (nil == identifier || [existingAccountNames containsObject:identifier]) {
         u_int32_t randomNumber = arc4random();
-        identifier = [NSString stringWithFormat:@"%@-%u", self.oauthClientId, randomNumber];
+        identifier = [NSString stringWithFormat:@"%@-%u", clientId, randomNumber];
     }
 
     return identifier;
 }
 
-- (SFUserAccount*)createUserAccount {
-    SFUserAccount *newAcct = [[SFUserAccount alloc] initWithIdentifier:[self uniqueUserAccountIdentifier]];
+
+- (SFUserAccount*)createUserAccount:(SFOAuthCredentials *)credentials {
+
+    SFUserAccount *newAcct = [[SFUserAccount alloc] initWithCredentials:credentials];
     SFOAuthCredentials *creds = newAcct.credentials;
```

Comment:
Lines 242-246 should be removed.  This method should not be manipulating the credentials passed in.